uniffi/src/did/did_resolver_adapter.rs

```diff
@@ -0,0 +1,44 @@
+use std::{
+    cell::RefCell,
+    sync::{Arc, Mutex},
+};
+
+use didcomm_core::{
+    did::{DIDDoc, DIDResolver as _DIDResolver},
+    error::{ErrorKind, Result, ResultExt},
+};
+use futures::channel::oneshot;
+
+use crate::{did_resolver::OnDIDResolverResult, DIDResolver};
+
+use async_trait::async_trait;
+
+pub(crate) struct DIDResolverAdapter {
```

Comment:
`DIDResolverAdapter` a bit ambitious name. As usually we use `{Type}DIDResolver` pattern. Maybe FFIDIDResolver?

Reply:
It's called Adapter as this is actually an adapter pattern: it adapts didcomm-uniffi DIDResolver to didcomm-core DIDResolver.
I don't think that `FFIDIDResolver` is much better or gives any clear insights what it does.